File: hw/tipan/u3/hw28Strategy/StrategyPython/model/insertion_sort.py
from model.sorting_strategy import SortingStrategy


class InsertionSort(SortingStrategy):

    def sort(self, data):

        array = data.copy()

        for i in range(1, len(array)):

            key = array[i]
            j = i - 1

            while j >= 0 and array[j] > key:
                array[j + 1] = array[j]
                j -= 1

            array[j + 1] = key

        return array

File: exams/palomo/u3/ExamP3/Strategy/src/ec/edu/espe/Strategy/model/InsertionSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.Strategy.model;

/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public class InsertionSort implements SortingStrategy{
    
    @Override
    public int[] sort(int[] data) {
        if (data == null) {
            return new int[0];
        }
        int[] sorted = data.clone();
        int n = sorted.length;
        for (int i = 1; i < n; i++) {
            int key = sorted[i];
            int j = i - 1;
            while (j >= 0 && sorted[j] > key) {
                sorted[j + 1] = sorted[j];
                j = j - 1;
            }
            sorted[j + 1] = key;
        }
        return sorted;
    }
}
